pkg/master/controller.go

```diff
@@ -301,20 +306,35 @@ func (r *masterCountEndpointReconciler) ReconcileEndpoints(serviceName string, i
 			Addresses: []api.EndpointAddress{{IP: ip.String()}},
 			Ports:     endpointPorts,
 		}}
-		return r.endpointRegistry.UpdateEndpoints(ctx, e)
+		return r.endpointRegistry.UpdateEndpoints(ctxDefault, e)
+	}
+
+	// Determine master count.
+	masterCount := r.masterCount
+
+	ctxSystem := api.WithNamespace(api.NewContext(), api.NamespaceSystem)
+	m, err := r.configmapRegistry.GetConfigMap(ctxSystem, componentconfig.ApiServerConfigMapName)
+	if err == nil {
+		if str, ok := m.Data[componentconfig.ApiServerConfigMapKey]; ok {
+			var config componentconfig.ApiServerConfiguration
+			err := json.Unmarshal([]byte(str), &config)
```

Comment:
Is it the proper way to parse our object? Where would conversion happen? I think we should use Decoder from `runtime/interfaces.go`. There are some examples how to do it.